pkg/controller/releasemanager/plan/syncApp.go

```diff
@@ -342,15 +316,42 @@ func (p *SyncApp) virtualService(log logr.Logger) *istiov1alpha3.VirtualService
 		return nil
 	}
 
+	hostsMap := map[string]bool{
+		p.serviceHost(): true,
+	}
+	for _, httpRoute := range http {
+		for _, httpMatch := range httpRoute.Match {
+			if httpMatch.Authority == nil || httpMatch.Authority.Prefix == "" {
+				continue
+			}
+			hostsMap[httpMatch.Authority.Prefix] = true
+		}
+	}
+	hosts := make([]string, len(hostsMap))
+	var i int
+	for host := range hostsMap {
+		hosts[i] = host
+		i++
+	}
```

Comment:
```suggestion
	hosts := make([]string, 0, len(hostsMap))
	for host := range hostsMap {
		hosts = append(hosts, host)
	}
```
save the explicit index management